docs/source/user_guide/gcm_based_inference/introduction.rst

```diff
@@ -181,5 +170,69 @@ This intervention says: "I'll ignore any causal effects of X on Y, and set every
 to 2.34." So the distribution of X will remain unchanged, whereas values of Y will be at a fixed
 value and Z will respond according to its causal model.
 
+These are the basic steps that need to happen. While we can run these steps explicitly, often they get
+executed as part of other steps, e.g. when fitting and re-fitting as part of computing confidence
+intervals. The next section therefore dives into a more typical usage pattern of the ``dowhy.gcm`` package.
+
+Typical usage of the ``dowhy.gcm`` package
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In practice, we may not execute the steps we've learned above explicitly and in this order, but they
+get called by other abstractions. E.g. we usually want to use confidence intervals, when answering
+causal questions, to quantify our confidence in the results. In this case, ``fit`` will be called on
+behalf of us, but we won't call it directly.
+
+At the same time, we may introduce some optional steps, again to improve our confidence in the
+results, such as some basic checks to see whether the independence structure of the graph is
+consistent with the data.
+
+Modeling an SCM and validating the structure of its causal graph against the data
+---------------------------------------------------------------------------------
+
+The graph modeling is the same as in `Step 1: Modeling cause-effect relationships as a structural
+causal model (SCM)`_. First we construct the causal graph:
+
+>>> causal_model = gcm.StructuralCausalModel(nx.DiGraph([('X', 'Y'), ('Y', 'Z')])) # X → Y → Z
+
+Now, we test this graph against our data:
+
+>>> rejection_result, _ = gcm.refute_causal_structure(causal_model.graph, data)
+>>> if rejection_result == gcm.RejectionResult.REJECTED:
+>>>      print("Do not continue. We advice revising the causal graph.")
+
+.. TODO: Add: For more details on structure validation and how to treat its results, see chapter
+:doc:`validating_causal_structure`.
+
+Unless the rejection result is ``REJECTED``, we continue to assign causal models:
+
+>>> gcm.auto.assign_causal_mechanisms(causal_model, based_on=data, quality=gcm.auto.AssignmentQuality.GOOD)
+
+Answering a causal query with confidence intervals
+--------------------------------------------------
+
+When we answer a causal query without computing its confidence intervals, what we effectively get,
+are point estimates. These are not very useful when trying to assess the confidence in our results.
+Instead of calling ``fit`` explicitly, we can achieve its execution by going through the API for
+confidence intervals. Let's say we wanted to understand the direct arrow strengths between nodes and
+quantify our confidence in those results. This is how we would do it:
+
+>>> strength_median, strength_intervals = gcm.confidence_intervals(
+>>>     gcm.bootstrap_training_and_sampling(gcm.direct_arrow_strength,
```

Comment:
(Not for this PR): What about renaming `bootstrap_training_and_sampling` to `fit_and_evaluate_subset` or something along the line? For me, this makes it clearer what it actually does. The multiple samples based on different data subsets are actually due to `confidence_intervals` calling it multiple times.

Reply:
Yea, I'm open to renaming that. We could even make it short and call it `fit_and_compute`. Then it would also nice read "fit and compute direct arrow strength" or similar.